doc-src/source/data-structure.rst

```diff
@@ -112,6 +113,135 @@ use the `rotate2` method::
+    around the Z-axis)
+
+  - pitch is positive when the x-axis pitches up (this is *opposite* the
+    right-hand-rule around the Y-axis)
+
+  - roll is positive according to the right-hand-rule around the
+    instument's x-axis
+
+Considerable care has been taken to make sure
+that these definitions of *heading*, *pitch*, *roll* and *orientmat*
+are consistent within |dlfn| between instrument models.
+However, because the instrument manufacturer's definitions of these
+variables are not consistent between instrument makes/models, this
+means that |dlfn|\ 's consistent definitions are often different from
+the definitions provided by an instrument manufacturer (i.e., there is
+no consensus on these definitions, so |dlfn| uses the above
+definitions).
+
+So, while |dlfn| uses the instrument manufacturer's definition of the
+instrument coordinate system ("XYZ"), the details of how this relates
+to the 'earth' coordinate system, and how *pitch*, *roll*, *heading*
+are computed are often distinct from the definitions specified by
+the instrument manufacturer (e.g., some manufacturers reference
+heading off of the y-axis rather than x, some use a 'north-east-down'
+earth reference frame, etc.). For practical purposes, when utilizing
+|dlfn| orientation data (for all instrument types), this all means
+that the user should:
+
+  - Use the instrument manufacturers definitions of XYZ
+
+  - Interpret *heading*, *pitch*, and *roll* data according to the above
+    definitions (ignore manufacturer definitions of these variables)
+
+  - All rotations into the earth frame will yield vectors that are in
+    a ENU coordinate system
+
+It is also possible to obtain the raw *heading*, *pitch*, *roll* data
+as defined by the instrument manufacturer using the
+``keep_orient_raw`` argument in the :func:`dolfyn.read` function. That
+data is contained in ``dat['orient']['raw']``. Note that this data
+does not obey the above definitions, and instead obeys the instrument
+manufacturer's definitions of these variables (i.e., it is exactly the
+data contained in the binary file). Also note that
+``dat['orient']['raw']['heading']`` is unaffected by setting declination
+as described in the next section.
+    
+Declination Handling
+....................
+
+|dlfn| includes functionality for handling `declination
+<https://www.ngdc.noaa.gov/geomag/declination.shtml>`_, but the value
+of the declination must be specified by the user. There are two ways
+to set a data-object's declination:
+
+1. Set declination explicitly using the ``dat.set_declination``
+   method, for example::
+
+     dat.set_declination(16.53)
+
+2. Set declination in the ``<data_filename>.userdata.json`` file
+   (`more details <json-userdata>`_ ), then read the binary data
+   file (i.e., using ``dat = dolfyn.read(<data_filename>)``).
+
+Both of these approaches produce modify the ``dat`` as described in
+the documentation for :meth:`~dolfyn.Velocity.set_declination` .
```

Comment:
Is this displaying the way you want it?
:meth:`~dolfyn.Velocity.set_declination`

Reply:
Will fix in my heading-bug-fix-mcf2 branch. Expect pull request in future.